repo/blob/gcs/gcs_storage.go

```diff
@@ -273,38 +251,28 @@ func tokenSourceFromCredentialsJSON(ctx context.Context, data json.RawMessage, s
 func New(ctx context.Context, opt *Options, isCreate bool) (blob.Storage, error) {
 	_ = isCreate
 
-	var ts oauth2.TokenSource
-
-	var err error
+	if opt.BucketName == "" {
+		return nil, errors.New("bucket name must be specified")
+	}
 
 	scope := gcsclient.ScopeReadWrite
```

Comment:
to support GCS blob extensions via `runFullMaintenance` it requires `gcsclient.ScopeFullControl` here

Reply:
Already addressed in #4151